docs/creating-a-runtime-check.md

```diff
@@ -0,0 +1,209 @@
+# Creating a Runtime Check
+
+Runtime checks are used to perform tests by executing the plugin's code.
+
+The Plugin Checker uses a number of interfaces to build out runtime checks.
+
+The Plugin Checker provides the `Runtime_Check` interface, which is used to identify a runtime check. This interface does not contain any methods but serves as a marker for runtime checks.
+
+```php
+use WordPress\Plugin_Check\Checker\Check_Result;
+use WordPress\Plugin_Check\Checker\Runtime_Check;
+use WordPress\Plugin_Check\Traits\Stable_Check;
+
+class Custom_Check extends Runtime_Check {
+  
+  use Stable_Check;
+  
+  public function get_categories() {
+    // Return an array of check categories.
+  }
+  
+  public function run( Check_Result $result );
+    // Handle running the check and adding warnings or errors to the result.
+  }
+}
+```
+
+In addtion, there is also the `Preparation` interface, which is used to define checks that have preparations. This interface defines a `prepare()` method which is used to run the logic required to prepare the environment before the check before is run.
+
+Both these interfaces are implemented in the `Abstract_Runtime_Check` class, which developers can use when building out their own runtime checks. This class defines both the `prepare()` and `run()` methods which are required to be implemented for every runtime check.
+
+Below is the basic scaffold when creating a custom runtime check.
+
+```php
+use WordPress\Plugin_Check\Checker\Check_Result;
+use WordPress\Plugin_Check\Checker\Checks\Abstract_Runtime_Check;
+use WordPress\Plugin_Check\Traits\Stable_Check;
+
+class Custom_Check extends Abstract_Runtime_Check {
+
+  use Stable_Check;
+  
+  public function get_categories() {
+    // Return an array of check categories.
+  }
+
+  public function prepare() {
+    // Handle the Checks preparations and return a cleanup function.
+  }
+
+  public function run( Check_Result $result ) {
+    // Handle running the check and adding warnings or errors to the result.
+  }
+}
+```
+
+## Preparations
+
+Preparations in the Plugin Checker are used to set up the environment before running checks, involving logic like activating themes or plugins and creating necessary test content. They utilize the `Preparation` interface, requiring the implementation of a `prepare()` method, and return cleanup functions to revert changes made during preparation. Three different approaches can be used for creating preparations based on the circumstances.
```

Comment:
See my previous comment, we shouldn't focus on activating themes and plugins here as this is too specific.

Also, before diving into the specific types of preparations, we should list them to clarify and connect the sections. Also, preparation classes and shared preparations are really referring to the same thing, so that is just one, not two approaches. On the other hand, the article is currently missing the concept of global preparations.
```suggestion
Preparations in the Plugin Checker are used to set up the environment before running checks, involving logic like adding actions and filters, and creating necessary test content. They utilize the `Preparation` interface, requiring the implementation of a `prepare()` method, and return cleanup functions to revert changes made during preparation.

Three different approaches can be used for creating preparations based on the circumstances:
* Check preparations
* Shared preparations
* Global preparations
```

Reply:
Thanks @felixarntz updated and added the new Global preparations section